lib/govuk_publishing_components/components/related_navigation_helper.rb

```diff
@@ -0,0 +1,245 @@
+class RelatedNavigationHelper
+  MAX_SECTION_LENGTH = 5
+  DEFINED_SECTIONS = %w(
+    related_guides
+    topics
+    collections
+    policies
+    topical_events
+    world_locations
+    statistical_data_sets
+  ).freeze
+
+  def initialize(content_item)
+    @content_item = content_item
+  end
+
+  def related_navigation
+    @related_content ||= [
+      { "related_items" => related_items },
+      { "related_guides" => related_guides },
+      { "collections" => related_collections },
+      { "topics" => related_topics },
+      { "policies" => related_policies },
+      { "topical_events" => related_topical_events },
+      { "world_locations" => related_world_locations },
+      { "statistical_data_sets" => related_statistical_data_sets },
+      { "worldwide_organisations" => related_worldwide_organisations },
+    ]
+
+    other = [related_external_links, related_contacts] || []
+    other.each do |sections|
+      sections.each do |section|
+        @related_content.push(
+          section["title"].tr(' ', '_') => section["links"]
+        )
+      end
+    end
+
+    @related_content
+  end
+
+  def other
+    @other ||= []
+  end
+
+  def construct_section_heading(section_title)
+    unless section_title === "related_items"
+      I18n.t('components.related_navigation.' + section_title, default: section_title.tr('_', ' '))
+    end
+  end
+
+  def section_css_class(css_class, section_title, link = {})
+    unless DEFINED_SECTIONS.include?(section_title) || link.fetch(:finder, false)
+      css_class += "--other"
+    end
+    css_class
+  end
+
+  def calculate_section_link_limit(links)
+    links.length == MAX_SECTION_LENGTH + 1 ? MAX_SECTION_LENGTH + 1 : MAX_SECTION_LENGTH
+  end
+
+  def remaining_link_count(links)
+    links.length - MAX_SECTION_LENGTH
+  end
+
+private
+
+  def build_links_for_sidebar(collection, path_key = "base_path", additional_attr = {})
+    collection.map do |link|
+      {
+        path: link[path_key],
+        text: link["title"]
+      }.merge(additional_attr)
+    end
+  end
+
+  def world_location_base_path(title)
+    "/world/#{parameterise(title)}/news"
+  end
+
+  def related_items
```

Comment:
The [finder links are currently appended to the `related_items` collection of links](https://github.com/alphagov/government-frontend/blob/master/app/presenters/navigation/finders.rb#L14). I don't think this should be a concern for this PR though as we are pretty doubtful of their efficacy and there's a good likelihood of their removal soon. If they are likely to be iterated into the related navigation component we can move the relevant presenter logic later.